### PR TITLE
Develop 4.0 integrate features

### PR DESCRIPTION
**Pull request by klevdiamanti**
# Title: Write a very short summary here
**Problem:** Write a description of your problem here.

**Solution:** Write a short summary of your solution.

**Key Features:**

* Key feature 1
* Key feature 2
* ...

## Checklist
- [ ] Make sure you are requesting to **pull a feature/bugfix branch** (right side). This should not be main or develop.
- [ ] Make sure you are make a pull request against either **main or develop** (left side). (Requesting to main should be reserved for bugfixes and new releases)
- [ ] Add or update unit tests (if applicable)
- [ ] Check your code with any unit tests (Run devtools::check() locally)
- [ ] Add neccessary documentation (if applicable)

## Type of changes

What type of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue) (link the issue on the right)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code style update (formatting, renaming)
- [ ] Documentation Update 
- [ ] Other (explain)

## Further comments
If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, any questions you have, etc...

Consider linking any issues (#issue-number ) or adding @mentions to ensure specific people see it.

### CODE REVIEW

File: OlinkAnalyze/R/olink_normalization.R [hunk 1]
```diff
@@ -596,35 +628,71 @@ norm_internal_cross_product <- function(ref_df,
...
-  # stores the outcome in df_norm
-  df_norm <- df_norm_bridge |>
+    dplyr::ungroup() |>
+    # add bridge normalized values
+    dplyr::left_join(
+      df_norm_bridge,
+      by = c(ref_cols$sample_id, ref_cols$olink_id, "Project"),
+      relationship = "one-to-one"
+    ) |>
+    # add bridgeable
```
Comment (MasoumehSheikhi) [inline OlinkAnalyze/R/olink_normalization.R:672]: Can we move this part after df_norm_qq, so the order of the columns be "MedianCenteredNPX"   "QSNormalizedNPX"   "BridgingRecommendation"?

File: OlinkAnalyze/R/olink_normalization.R
Comment (MasoumehSheikhi) [inline OlinkAnalyze/R/olink_normalization.R:1]: line [183] It doesn't let me comment on unchanged parts. 
 Add an example to olink_normalization() for cross product bridging 

# Cross-product normalization:
# Find overlapping samples, but exclude Olink control
overlap_samples <- intersect((data_ht |>
                                dplyr::filter(SampleType == "SAMPLE"))$SampleID,
                             (data_3k |>
                                dplyr::filter(SampleType == "SAMPLE"))$SampleID)
# Normalize
olink_normalization(
  df1 = data_ht,
  df2 = data_3k,
  overlapping_samples_df1 = overlap_samples,
  df1_project_nr = 'P1',
  df2_project_nr = 'P2',
  reference_project = 'P1'
)

**Review by MasoumehSheikhi — CHANGES_REQUESTED** [review]
Please see comments about
- adding itemized detail
- adding example
- reordering the columns

File: OlinkAnalyze/R/olink_normalization.R (hunk 1 above)
Comment (klevdiamanti) [reply OlinkAnalyze/R/olink_normalization.R:672]: I guess you mean joining the results with the following order:
1. add bridge normalized
2. add QS normalized
3. add is bridgeable.
Fixed.

File: OlinkAnalyze/R/olink_normalization.R [hunk 2]
```diff
@@ -180,6 +205,25 @@
...
+#' # get reference samples
+#' overlap_samples_product <- intersect(
+#'   x = unique(OlinkAnalyze:::data_ht$SampleID),
+#'   y = unique(OlinkAnalyze:::data_3k$SampleID)
+#' ) |>
+#'   (\(.) .[!grepl("CONTROL", .)])()
+#'
+#' # normalize
+#' olink_normalization(
+#'   df1 = OlinkAnalyze:::data_ht,
```
Comment (MasoumehSheikhi) [inline OlinkAnalyze/R/olink_normalization.R:220]: after merging PR #454, do we need to change instances like OlinkAnalyze:::data_ht to OlinkAnalyze:::data_ht_small?

**Pull request merged**